torch_xla/csrc/tensor.cpp

```diff
@@ -219,18 +227,19 @@ torch::lazy::Value IrValueFromScalar(const at::Scalar& value,
                                      const torch::lazy::BackendDevice& device,
                                      bool transfer_async) {
   at::Tensor tensor = at::scalar_tensor(value, at::TensorOptions(scalar_type));
-  torch::lazy::BackendDataPtr device_data =
-      TensorToXlaData(tensor, device, transfer_async);
+  torch::lazy::BackendDataPtr device_data = TensorToXlaData(
+      tensor, device, transfer_async || ShouldTransferScalarAsync());
   return torch::lazy::MakeNode<DeviceData>(std::move(device_data));
 }
 
 torch::lazy::BackendDataPtr GetDeviceData(
-    const at::Tensor& tensor, const torch::lazy::BackendDevice& device) {
+    const at::Tensor& tensor, const torch::lazy::BackendDevice& device,
+    bool transfer_async = false) {
   XlaDataCacheArena::XlaDataCache* cache = GetXlaDataCache(device);
   torch::lazy::BackendDataPtr device_data = cache->Get(tensor);
   if (device_data == nullptr) {
     at::Tensor tensor_copy = torch::lazy::CopyTensor(tensor);
-    device_data = TensorToXlaData(tensor_copy, device);
+    device_data = TensorToXlaData(tensor_copy, device, transfer_async);
```

Comment:
Let's move all of these logic (check env var) into `TensorToXlaData` instead of methods in `tensor.cpp`. The only reason I do it this way is I can't tell which call to `TensorToXlaData` is from RNG seed but if we want to transfer all scalar async then this additional layer is not necessary.

Reply:
I also don't see any particular scalar check(check whether a tensor is scalar before deciding it should be send async)

Reply:
We should also kill the RNG option if this `XLA_TRANSFER_SCALAR_ASYNC` covers everything you need.

Reply:
If we move the check to TensorToXLAData, we can kill RNG option and have two checks: 1) Env variable set check 2) If scalar check.
I can try to make those work